src/plasmapy/formulary/collisions/misc.py

```diff
@@ -325,6 +464,213 @@ def Bethe_stopping(
+
+    Parameters
+    ----------
+    m : `~astropy.units.Quantity`
+        The mass of the projectile specie.
+
+    v : `~astropy.units.Quantity`
+        The speed of the projectile particles streaming through the target.
+
+    z : `~astropy.units.Quantity`
+        The atomic number of the projectile specie.
+
+    A : `~astropy.units.Quantity`
+        The atomic weight of the target material in units convertible to kilograms per mol.
+
+    Rho : `~astropy.units.Quantity`
+        The density of the target in units convertible to kilograms per cubic meter.
+
+    Z : `~astropy.units.Quantity`
+        The atomic number of the target.
+
+    t : `~astropy.units.Quantity`
+        The thickness of the target in units convertible to meters.
+
+    return_rms : `bool`
+        Whether to return the rms scattering angle in radians, or return the
+        distribution function.
+
+    """
+    beta = v / _c
+    gamma = 1 / np.sqrt(1 - beta**2)
+    p = gamma * m * v
+
+    # Number density of atoms in the target
+    N = Rho * _N_A / A
+
+    # Eq. 10 with relativistically correct `p`
+    χ_c = np.sqrt(
+        4 * np.pi * N * t * const.e.esu**4 * Z * (Z + 1) * z**2 / (p * v) ** 2
+    )
+
+    # Eq. 21a, "the deviation from the Born approximation"
+    Born_alpha = z * Z * const.e.esu**2 / (_hbar * v)
+
+    # Eq. 22, collected constants have been recalculated. Bethe switches to the
+    # target areal density for `t`. We instead introduce a factor of `rho` and
+    # keep `t` as the target thickness.
+    e_b = (
+        (6702 * u.cm**2 / u.mol)
+        * Rho
+        * t
+        / beta**2
+        * (Z + 1)
+        * Z ** (1 / 3)
+        * z**2
+        / (A * (1 + 3.34 * Born_alpha**2))
+    )
+    b = np.log(e_b.cgs.value)
+
+    def Moliere_scattering_B(B):
+        """Eq. 23 of Bethe."""
+        return B - np.log(B) - b
+
+    # The transcendental equation associated with `B` yields two solutions for
+    # every `b`. We want to solve for values where B > 1, this corresponds to
+    # our initial guess satisfying b > 1.
+    B = fsolve(
+        Moliere_scattering_B,
+        x0=np.full_like(b, 5),
+    )
+
+    def scattering_integrand(theta):
+        """Eq. 25 of Bethe."""
+
+        # Eq. 24 of Bethe
+        ϑ = theta / (χ_c * np.sqrt(B))
+
+        if np.any(ϑ > 20):
+            raise PhysicsError(f"Exceeded ϑ=20 (got {np.max(ϑ.cgs)}).")
+
+        if np.any(theta > np.pi):
+            raise PhysicsError(
+                f"Angular distribution cannot exceed theta=180 (got {np.max(theta)})."
+            )
+
+        B_coefficient = np.asarray([1 / B**i for i in range(3)])
+
+        if use_interpolator:
+            f_n = _f_mol_spline(ϑ.cgs).T
+        else:
+            f_n = [_f_mol_n(ϑ, i) for i in range(3)]
+
+        return np.sum(B_coefficient * f_n, axis=0)
+
+    if not return_rms:
+        return scattering_integrand
+    else:
+        # Eq. 30
+        theta_w = χ_c * np.sqrt(B)
```

Comment:
Why not just use pi as the upper bound, and not rely on this approximation that neglects the higher order terms?

Reply:
TODO: Figure out heuristic for including higher-order corrections when necessary